services/github/github-api-provider.js

```diff
@@ -185,7 +185,8 @@ class GithubApiProvider {
         baseUrl,
         headers: {
           'User-Agent': 'Shields.io',
-          Accept: 'application/vnd.github.v3+json',
+          Accept:
+            'application/vnd.github.v3+json, application/vnd.github.antiope-preview+json',
           Authorization: `token ${tokenString}`,
         },
```

Comment:
Was the overriding of provided options deliberate here? I had to add the alternative media type required for this API here as it wasn't working from the service class due to the merge order

Reply:
I'm sure there is an open issue about the way we're merging options, but I can't find it for the life of me.

We often merge in the other order e.g: https://github.com/badges/shields/blob/c9cd77a7468f495575743fbb5d37ddd217c91098/core/base-service/base-json.js#L43-L46

but the downside of this is if we want to pass an extra header, we have to re-add `Accept: 'application/json'` (because the spread operation doesn't nest, if you see what I mean)

I suspect the reason we've done it in the other order here is so that if we pass any header, its not going to overwrite the `Authorization` header.

In general, we need a more sophisticated merge starategy for these `options` objects.